easybuild/framework/easyconfig/types.py

```diff
@@ -76,6 +78,16 @@ def is_value_of_type(value, typ_spec):
                 # check whether all values have allowed types
                 'value_types': lambda val: all([type(el) in extra_reqs['value_types'] for el in val.values()]),
```

Comment:
hmm, maybe we should make this a recursive call to `is_value_of_type` (which should be made aware of `EASY_TYPES`)?